lana/src/main/scala/com/financialforce/lana/SymbolFinder.scala

```diff
@@ -3,52 +3,32 @@
  */
 package com.financialforce.lana
 
-import com.nawforce.common.api.{Name, TypeName}
-import com.nawforce.common.diagnostics.CatchingLogger
-import com.nawforce.common.documents.DocumentIndex
-import com.nawforce.common.path.PathFactory
-import com.nawforce.common.sfdx.{MDAPIWorkspace, Project, SFDXWorkspace}
-
-import scala.collection.mutable
-import scala.concurrent.Future
-import scala.scalajs.concurrent.JSExecutionContext.Implicits.queue
 import scala.scalajs.js
-import scala.scalajs.js.JSConverters._
+import scala.scalajs.js.annotation.JSImport
+
+@js.native
+@JSImport("pkgforce", "Workspaces")
+object Workspaces extends js.Object {
+  def get(path: String): Workspace = js.native
+}
+
+@js.native
+@JSImport("pkgforce", "Workspace")
+class Workspace extends js.Object {
+  def findType(typeName: String): String = js.native
+}
 
 class SymbolFinderError(err: String) extends Exception(err)
 
 class SymbolFinder {
-  private val workspaceDocumentIndexes = new mutable.HashMap[String, DocumentIndex]()
 
-  def findSymbol(wsPath: String, symbol: String): js.Promise[String] = {
-    val index = getIndex(wsPath)
-    val typeName = TypeName(symbol.split('.').take(2).map(Name(_)).reverse)
-    val result = index.getByType(typeName) match {
+  def findSymbol(wsPath: String, symbol: String): String = {
+    val ws = Workspaces.get(wsPath)
+    Option(ws.findType(symbol)) match {
       case None =>
-        Future.failed(new SymbolFinderError(s"Symbol $symbol not found"))
-      case Some(metadataDocument) =>
-        Future.successful(metadataDocument.path.toString)
+        throw new SymbolFinderError(s"Symbol $symbol not found")
+      case Some(path) =>
+        path
     }
```

Comment:
Is there any difference between this being async and now not, noticed an issue or just simplifying?

Reply:
I originally wrote as a possible extension point so thought returning js.Promise made most sense in case we needed async behaviour. I don't think we are going to use that now so I simplified back to sync to make a little easier to understand.

Reply:
Thanks for reviewing :-)